src/cache/cache-memory.ts

```diff
@@ -0,0 +1,43 @@
+import { ICache, CacheEntry, CacheKey, findExistingCacheKey } from './shared';
+
+export class InMemoryCache {
+  public enclosedCache: ICache = (function () {
+    let cache: Record<string, unknown> = {};
+
+    return {
+      set(key: string, entry: unknown): Promise<void> {
+        cache[key] = entry;
+        return Promise.resolve();
+      },
+
+      get(key: string): Promise<Partial<CacheEntry> | undefined> {
```

Comment:
The signature here differs from the LocalStorage one, wouldnt it make sense to have this in line?

The local storage one has 
```
public get(key: string): Promise<unknown> {
```
whereas the in memory uses

```
get(key: string): Promise<Partial<CacheEntry> | undefined> {
```
This makes the local storage one usable for types other then CacheEntry, but the In memory not.

Would it make sense to apply generics to both and change both to be:
```
get<TData = Unknown>(key: string): Promise<TData | undefined> {
```

Reply:
Good catch, those signatures should more than likely be the same - I'll take a look.